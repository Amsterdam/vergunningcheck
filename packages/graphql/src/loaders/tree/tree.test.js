require("@spotify/polly-jest-presets");
const db = require("../../database");

const tree = require(".");

tree.search = jest.fn(tree.search);
tree.fetch = jest.fn(tree.fetch);
tree.reducer = jest.fn(tree.reducer);

const id1 =
  "G0363.2822a91bbdd840e48086103228f55f08~20051223175652~20160503164617";
const id2 =
  "G0363.376d57e9f24649db8d2a03ff996c2c1d~20051223175629~20160503164617";

afterAll(() => {
  db.close();
});

describe("Tree loader", () => {
  test("tree.reducer", async () => {
    expect(
      tree.reducer({
        id: "123abc",
        nonsupported: "gone",
        geometry:
          '{"type":"Point", "coordinates":[52.3707669432068,4.90550562949144]}',
      })
    ).toMatchObject({
      id: "123abc",
      geometry: {
        type: "Point",
        coordinates: [52.3707669432068, 4.90550562949144],
      },
    });
  });

  test("tree.search", async () => {
    const geoJson = {
      type: "Polygon",
      coordinates: [
        [
          [4.903454, 52.370005],
          [4.905888, 52.370005],
          [4.905888, 52.370821],
        ],
      ],
    };
    expect(await tree.search(geoJson)).toMatchSnapshot();
  });

  describe("tree.fetch", () => {
    test("single fetch", async () =>
      expect(await tree.fetch([id1])).toMatchSnapshot());

    test("multi-fetch", async () =>
      expect(await tree.fetch([id1, id2])).toMatchSnapshot());
  });
});
