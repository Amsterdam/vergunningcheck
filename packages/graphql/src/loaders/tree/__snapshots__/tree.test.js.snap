// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree loader tree.fetch multi-fetch 1`] = `
Array [
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3707669432068,
        4.90550562949144,
      ],
      "type": "Point",
    },
    "id": "G0363.2822a91bbdd840e48086103228f55f08~20051223175652~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3702637443156,
        4.9043153028201,
      ],
      "type": "Point",
    },
    "id": "G0363.376d57e9f24649db8d2a03ff996c2c1d~20051223175629~20160503164617",
  },
]
`;

exports[`Tree loader tree.fetch single fetch 1`] = `
Array [
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3707669432068,
        4.90550562949144,
      ],
      "type": "Point",
    },
    "id": "G0363.2822a91bbdd840e48086103228f55f08~20051223175652~20160503164617",
  },
]
`;

exports[`Tree loader tree.search 1`] = `
Array [
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3706736230273,
        4.90402531512053,
      ],
      "type": "Point",
    },
    "id": "G0363.002591fcc2b84e87aa742ae8955704f6~20051223175647~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3701168980927,
        4.90397039814735,
      ],
      "type": "Point",
    },
    "id": "G0363.0f4606afae694fc7a872f96a83c6e8e8~20051223175637~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3707669432068,
        4.90550562949144,
      ],
      "type": "Point",
    },
    "id": "G0363.2822a91bbdd840e48086103228f55f08~20051223175652~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.370438514487,
        4.90428768918262,
      ],
      "type": "Point",
    },
    "id": "G0363.33744e2c56124a0caa5b7dfe3fc80bda~20051223175629~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3702637443156,
        4.9043153028201,
      ],
      "type": "Point",
    },
    "id": "G0363.376d57e9f24649db8d2a03ff996c2c1d~20051223175629~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3705954604083,
        4.90411206637859,
      ],
      "type": "Point",
    },
    "id": "G0363.396127a9015f404795b5acc3808568d1~20051223175647~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3704747170414,
        4.90481245762887,
      ],
      "type": "Point",
    },
    "id": "G0363.5a86b408d75c4f6288cd9217e67d594b~20051223175630~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.370695240745,
        4.90533369756735,
      ],
      "type": "Point",
    },
    "id": "G0363.62f2518359024b09ad2d38788956c922~20051223175651~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3705161377,
        4.90420087099772,
      ],
      "type": "Point",
    },
    "id": "G0363.64443347893a4462a2a64c5643e15f33~20051223175630~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3705906393038,
        4.90352556220802,
      ],
      "type": "Point",
    },
    "id": "G0363.657f35ef3da144c9b0044a9cab8b744d~20051223175647~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3703942559505,
        4.90462200433021,
      ],
      "type": "Point",
    },
    "id": "G0363.68b256f073234bbba629491d6059fe56~20051223175630~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.370819506211,
        4.90562123704666,
      ],
      "type": "Point",
    },
    "id": "G0363.6f39e985f0f046b0bede049682457cbe~20051223175651~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3703697951395,
        4.90436498323601,
      ],
      "type": "Point",
    },
    "id": "G0363.8ef5add18e6b437c96df364ff00d84e2~20051223175629~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3707573580705,
        4.90393859095072,
      ],
      "type": "Point",
    },
    "id": "G0363.93ab39325d6e4df5afc4b02c105731f5~20051223175647~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3706987106787,
        4.90378603404174,
      ],
      "type": "Point",
    },
    "id": "G0363.9916a75449174e709097c7eeaa27d30a~20051223175645~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3700488668313,
        4.90380885652741,
      ],
      "type": "Point",
    },
    "id": "G0363.abc612c85a1348da98c3e42f4da4c293~20051223175637~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3706333381144,
        4.90518469851331,
      ],
      "type": "Point",
    },
    "id": "G0363.b64a9908664f47a39a08cb0431ee8cbc~20051223175651~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3701933737181,
        4.90414421190167,
      ],
      "type": "Point",
    },
    "id": "G0363.bda5f77346464795bf10b8418d3d5a90~20051223175629~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3706476509254,
        4.90366042762065,
      ],
      "type": "Point",
    },
    "id": "G0363.c218196eb22949ca90e453953990ee5a~20051223175647~20160503164617",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        52.3705593780738,
        4.90501026656054,
      ],
      "type": "Point",
    },
    "id": "G0363.ea659d45c6b94a90bbfcf17697f6afba~20051223175650~20160503164617",
  },
]
`;
